### ISSUE TITLE
MAAS config

### ISSUE THREAD

**Issue by zarez** [andyshinn/kuberdoo#1]
Hi Andy,

I tried to install and configure maas to node provisioning without much success. I did set next-server on hex to  maas node(192.168.33.11) . Set maas with forwarding DNS to router. Looks like have to open the kuber to see whats posting on boot. Any tips are appreciated. 

Cheers,
Gjorgji

**Comment by zarez** [comment]
Found the issue: set boot-file-name=pxelinux.0( it was netboot.xyz.kpxe)
Now maas has the node but I cannot not commission due to power configuration. what did you pick up for power type?

**Comment by zarez** [comment]
Set to manual. Now its commissioned and in Ready state. Will play more tomorrow.

**Comment by andyshinn** [comment]
Yep, setting to manual is the correct way for these boards. When I was using MAAS it looks something like this:

* Commision node
* Hit power button
* Wait for power light to go back to yellow
* Provision node
* Hit power button

The config in this repository for the MikroTik probably doesn't have MAAS as the default. I had a couple different options to do preseed Ubuntu or netboot.xyz. I would switch between them doing something like:

```
/ip dhcp-server lease> set [ find dhcp-option ~ "udoo*" ] dhcp-option-set=boot-preseed
```

Then when they are done booting I would turn off PXE boot with:

```
/ip dhcp-server lease set [ find dhcp-option ~ "udoo*" ] dhcp-option-set=none
```

Hope that helps!

**Comment by zarez** [comment]
I just changed the next-server to maas IP and the boot-file-name to pxelinux.0. Deployed the nodes from maas. The only problem was hostnames as for some reason maas was using some of his own, not the one from dhcp(hex). 
What deployment did you use for kubernetes?

**Comment by andyshinn** [comment]
MAAS will always generate a hostname. You need to click on the hostname to edit it after it shows up in MAAS (you can edit is before comissioning). After that, it will stay tied to that MAC address in MAAS.

I used Kubespray (https://github.com/kubernetes-incubator/kubespray) to build my Kubernetes cluster. I think it is the easiest to use for own hardware. My inventory file looked something like this (change your IP to match if using different subnet):

```
udoo1 ansible_host=192.168.88.11
udoo2 ansible_host=192.168.88.12
udoo3 ansible_host=192.168.88.13
udoo4 ansible_host=192.168.88.14
udoo5 ansible_host=192.168.88.15
udoo6 ansible_host=192.168.88.16

[kube-master]
udoo1
udoo2

[etcd]
udoo1
udoo2
udoo3

[kube-node]
udoo1
udoo2
udoo3
udoo4
udoo5
udoo6

[k8s-cluster:children]
kube-node
kube-master
```

**Comment by zarez** [comment]
Yep, changed the name in the MAAS gui before deploying. 

Will check Kuberspray. One thing: as MAAS seats on one of the nodes (udoo1) I was planing to use as master, but if I need to provision the cluster more then one I guess it will be manual uninstall. Do you have the same setup? 

I was also thinking getting one more node in the case just for deployment purposes. I think that if you dont power all the nodes at one, one extra board will be ok for that power supply.
